IPython/core/interactiveshell.py

```diff
@@ -2615,6 +2621,7 @@ def run_cell(self, raw_cell, store_history=False, silent=False, shell_futures=Tr
         result : :class:`ExecutionResult`
         """
         result = ExecutionResult()
+        self.last_execution_succeed = True
```

Comment:
Resetting it at the start of `run_cell` means that code in the cell cannot use `get_ipython().last_execution_succeed`. Maybe that use case is not important, but it would be pretty surprising.